From: Youquan Song <youquan.song@intel.com>
Subject: x86/perf: Add IvyBridge EP support
Date: Tue Dec 18 12:20:23 2012 -0500
Git-commit: 923d8697e24847000490c187de1aeaca622611a3
References: bnc#805456
Patch-mainline: v3.8-rc7
Signed-off-by: Tony Jones <tonyj@suse.de>

    x86/perf: Add IvyBridge EP support
    
    Running the perf utility on a Ivybridge EP server we encounter
    "not supported" events:
    
       <not supported> L1-dcache-loads
       <not supported> L1-dcache-load-misses
       <not supported> L1-dcache-stores
       <not supported> L1-dcache-store-misses
       <not supported> L1-dcache-prefetches
       <not supported> L1-dcache-prefetch-misses
    
    This patch adds support for this processor.
    
    Signed-off-by: Youquan Song <youquan.song@intel.com>
    Cc: Andi Kleen <ak@linux.intel.com>
    Cc: Youquan Song <youquan.song@linux.intel.com>
    Cc: Peter Zijlstra <peterz@infradead.org>
    Cc: Stephane Eranian <eranian@google.com>
    Link: http://lkml.kernel.org/r/1355851223-27705-1-git-send-email-youquan.song@intel.com
    Signed-off-by: Ingo Molnar <mingo@kernel.org>

---
 arch/x86/kernel/cpu/perf_event_intel.c |    1 +
 1 file changed, 1 insertion(+)

--- a/arch/x86/kernel/cpu/perf_event_intel.c
+++ b/arch/x86/kernel/cpu/perf_event_intel.c
@@ -2070,6 +2070,7 @@ __init int intel_pmu_init(void)
 		pr_cont("SandyBridge events, ");
 		break;
 	case 58: /* IvyBridge */
+	case 62: /* IvyBridge EP */
 		memcpy(hw_cache_event_ids, snb_hw_cache_event_ids,
 		       sizeof(hw_cache_event_ids));
 		memcpy(hw_cache_extra_regs, snb_hw_cache_extra_regs,
